tests/test_check_digit_rust.py

```diff
@@ -0,0 +1,111 @@
+    assert not check_digit_rust_py.is_valid(resource_id_with_incorrect_character_and_correct_check_digit)
+
+    # reject a string with swapped characters
+    resource_id_with_swapped_characters = "cmfk1DMHRBiR4_-6HXpEFA"
+    resource_id_with_swapped_characters_and_correct_check_digit = resource_id_with_swapped_characters + correct_resource_id_check_digit
+    assert not check_digit_rust_py.is_valid(resource_id_with_swapped_characters_and_correct_check_digit)
+
+
+def test_rust_vs_python_parity():
+    """Test that Rust and Python implementations produce identical results."""
+    from ark_resolver import check_digit as check_digit_py
+
+    test_codes = [
+        "cmfk1DMHRBiR4-_6HXpEFA",
+        "ABC123",
+        "hello-world_test",
+        "0123456789",
+        "abcdefghijklmnopqrstuvwxyz",
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
+        "-_",
+    ]
+
+    for code in test_codes:
+        # Test calculate_check_digit parity
+        py_check_digit = check_digit_py.calculate_check_digit(code)
+        rust_check_digit = check_digit_rust_py.calculate_check_digit(code)
+        assert py_check_digit == rust_check_digit, (
+            f"Check digit mismatch for '{code}': Python='{py_check_digit}', Rust='{rust_check_digit}'"
+        )
+
+        # Test is_valid parity for code without check digit
+        py_valid_no_check = check_digit_py.is_valid(code)
+        rust_valid_no_check = check_digit_rust_py.is_valid(code)
+        assert py_valid_no_check == rust_valid_no_check, (
+            f"is_valid mismatch for '{code}' (no check): Python={py_valid_no_check}, Rust={rust_valid_no_check}"
+        )
+
+        # Test is_valid parity for code with check digit
+        code_with_check = code + py_check_digit
+        py_valid_with_check = check_digit_py.is_valid(code_with_check)
+        rust_valid_with_check = check_digit_rust_py.is_valid(code_with_check)
+        assert py_valid_with_check == rust_valid_with_check, (
+            f"is_valid mismatch for '{code_with_check}': Python={py_valid_with_check}, Rust={rust_valid_with_check}"
+        )
+
+        # Test calculate_modulus parity
+        py_modulus_no_check = check_digit_py.calculate_modulus(code, False)
+        rust_modulus_no_check = check_digit_rust_py.calculate_modulus(code, False)
+        assert py_modulus_no_check == rust_modulus_no_check, (
+            f"calculate_modulus mismatch for '{code}' (no check): Python={py_modulus_no_check}, Rust={rust_modulus_no_check}"
+        )
+
+        py_modulus_with_check = check_digit_py.calculate_modulus(code_with_check, True)
+        rust_modulus_with_check = check_digit_rust_py.calculate_modulus(code_with_check, True)
+        assert py_modulus_with_check == rust_modulus_with_check, (
+            f"calculate_modulus mismatch for '{code_with_check}' (with check): Python={py_modulus_with_check}, Rust={rust_modulus_with_check}"
+        )
```

Comment:
```suggestion
        assert py_modulus_with_check == rust_modulus_with_check, (
            f"calculate_modulus mismatch for '{code_with_check}' (with check): " +
            f"Python={py_modulus_with_check}, Rust={rust_modulus_with_check}"
        )
```

I think this is how you'd resolve the linting issue here, in Python